# Cycle 2 — Profitability & Loss Driver Analysis

This notebook analyses business profitability using a star schema built in Cycle 2.

In [1]:
import pandas as pd

In [2]:
category_margin = pd.read_csv("../data/processed/category_margin.csv")
loss_products = pd.read_csv("../data/processed/loss_products_by_category.csv")
loss_customers = pd.read_csv("../data/processed/loss_customers_summary.csv")
loss_customer_category = pd.read_csv("../data/processed/category_profit_loss_customers.csv")
discount_band = pd.read_csv("../data/processed/discount_band_margin.csv")

## Category Performance

This shows overall profitability by category.

In [3]:
category_margin

,category,revenue,profit,margin_pct
0,Technology,836154.10,145454.9481,17.40
1,Office Supplies,719046.99,122490.8008,17.04
2,Furniture,741999.98,18451.2728,2.49


Technology and Office Supplies show strong margins (~17%).

Furniture generates high revenue but has very low margin (~2.5%), indicating structural inefficiency.

## Loss-Making Products

This shows how many products lose money and where they are concentrated.

In [4]:
loss_products

,category,loss_products,total_loss
0,Furniture,120,-36910.8401
1,Technology,53,-27191.9666
2,Office Supplies,121,-12594.6887


Loss-making products are concentrated in Furniture, confirming it as a structurally weak category.

## Loss-Making Customers

In [5]:
loss_customers


,loss_customers,total_loss
0,155,-71224.4744


155 customers are unprofitable, contributing to a total loss of approximately -71,224.

## Category Performance for Loss Customers

In [6]:
loss_customer_category

,category,revenue,profit,margin_pct
0,Furniture,150638.09,-26323.3034,-17.47
1,Office Supplies,116385.27,-23475.6614,-20.17
2,Technology,124927.17,-21425.5096,-17.15


All categories become unprofitable when sold to loss-making customers, indicating customer-level behaviour (discounting) is a key issue.

## Discount Impact on Profitability

In [7]:
discount_band

,discount_band,order_lines,revenue,profit,margin_pct
0,0–10%,94,54369.30,9029.1770,16.61
1,0%,4798,1087908.47,320987.6032,29.51
2,10–20%,3709,792152.87,91756.2975,11.58
3,20–30%,227,103226.76,-10369.2774,-10.05
4,30–40%,233,130911.28,-25448.1881,-19.44
5,40%+,933,128632.39,-99558.5905,-77.40


Higher discount levels significantly reduce margins and often lead to losses.

## Final Insight

- Technology is the main profit driver  
- Office Supplies is stable  
- Furniture has weak margins  
- Losses are driven by:
  - specific products  
  - specific customers  
  - excessive discounting  

### Recommendations

- Limit discount levels  
- Review Furniture pricing and costs  
- Monitor customer profitability  